src/Spanner/Database.php

```diff
@@ -0,0 +1,1313 @@
+    {
+        // There isn't anything configurable here.
+        $options['transactionOptions'] = [
+            'readWrite' => []
+        ];
+
+        $session = $this->selectSession(SessionPoolInterface::CONTEXT_READWRITE);
+
+        try {
+            return $this->operation->transaction($session, $options);
+        } finally {
+            $session->setExpiration();
+        }
+    }
+
+    /**
+     * Insert a row.
+     *
+     * Mutations are committed in a single-use transaction.
+     *
+     * Example:
+     * ```
+     * $database->insert('Posts', [
+     *     'ID' => 1337,
+     *     'postTitle' => 'Hello World!',
+     *     'postContent' => 'Welcome to our site.'
+     * ]);
+     * ```
+     *
+     * @codingStandardsIgnoreStart
+     * @see https://cloud.google.com/spanner/reference/rpc/google.spanner.v1#google.spanner.v1.CommitRequest CommitRequest
+     * @codingStandardsIgnoreEnd
+     *
+     * @param string $table The table to mutate.
+     * @param array $data The row data to insert.
+     * @param array $options [optional] Configuration options.
+     * @return Timestamp The commit Timestamp.
+     */
+    public function insert($table, array $data, array $options = [])
+    {
+        return $this->insertBatch($table, [$data], $options);
+    }
+
+    /**
+     * Insert multiple rows.
+     *
+     * Mutations are committed in a single-use transaction.
+     *
+     * Example:
+     * ```
+     * $database->insertBatch('Posts', [
+     *     [
+     *         'ID' => 1337,
+     *         'postTitle' => 'Hello World!',
+     *         'postContent' => 'Welcome to our site.'
+     *     ], [
+     *         'ID' => 1338,
+     *         'postTitle' => 'Our History',
+     *         'postContent' => 'Lots of people ask about where we got started.'
+     *     ]
+     * ]);
+     * ```
+     *
+     * @codingStandardsIgnoreStart
+     * @see https://cloud.google.com/spanner/reference/rpc/google.spanner.v1#google.spanner.v1.CommitRequest CommitRequest
+     * @codingStandardsIgnoreEnd
+     *
+     * @param string $table The table to mutate.
+     * @param array $dataSet The row data to insert.
+     * @param array $options [optional] Configuration options.
+     * @return Timestamp The commit Timestamp.
+     */
+    public function insertBatch($table, array $dataSet, array $options = [])
+    {
+        $mutations = [];
+        foreach ($dataSet as $data) {
+            $mutations[] = $this->operation->mutation(Operation::OP_INSERT, $table, $data);
+        }
+
+        $session = $this->selectSession(SessionPoolInterface::CONTEXT_READWRITE);
+
+        $options['singleUseTransaction'] = $this->configureTransactionOptions();
+
+        try {
+            return $this->operation->commit($session, $mutations, $options);
+        } finally {
+            $session->setExpiration();
+        }
+    }
+
+    /**
+     * Update a row.
+     *
+     * Only data which you wish to update need be included. You must provide
```

Comment:
Why not just use the text in the [rpc](https://cloud.google.com/spanner/docs/reference/rpc/google.spanner.v1#google.spanner.v1.Mutation.Write):

"The list of columns must contain enough columns to allow Cloud Spanner to derive values for all primary key columns in the row(s) to be modified."